crates/uv-python/src/environment.rs

```diff
@@ -154,20 +172,24 @@ impl PythonEnvironment {
         if venv.is_file() {
             return Err(InvalidEnvironment {
                 path: venv,
-                reason: "expected directory but found a file".to_string(),
+                kind: InvalidEnvironmentKind::NotDirectory,
             }
             .into());
         }
 
-        if !venv.join("pyvenv.cfg").is_file() {
+        let executable = virtualenv_python_executable(&venv);
+
+        // Check if the executable exists before querying so we can provide a more specific error
+        // Note we intentionally don't require a resolved link to exist here, we're just trying to
+        // tell if this _looks_ like a Python environment.
+        if !(executable.is_symlink() || executable.is_file()) {
             return Err(InvalidEnvironment {
                 path: venv,
-                reason: "missing a `pyvenv.cfg` marker".to_string(),
+                kind: InvalidEnvironmentKind::MissingExecutable(executable.clone()),
             }
             .into());
-        }
+        };
```

Comment:
Would this mean that a ~/ where ~/bin/python exists is an environment?

Reply:
Uhh yeah. Which is could be problematic yes. And `/` where `/bin/python` exists. 

Unfortunately I can't think of a better way to tell if an arbitrary directory is a system Python environment?

Reply:
Okay I think I've guarded against this now by only deleting virtual environments.